### PR TITLE
Don't throw in `GetProposedDimensions`

### LINKED ISSUE #10211: Startup Crash in _HandleCreateWindow on 1.8 Stable
### Windows Terminal version (or Windows build number)

1.8.144444

### Other Software

_No response_

### Steps to reproduce

No idea.

### Expected Behavior

_No response_

### Actual Behavior

Look at Watson bucket `4ba5cd02-f285-99e4-6254-56f3218146c9` or failure ID `5133fdab-7a9f-5853-f23d-43d441146a6c`. 3,000 hits in the last few days.

### PR DESCRIPTION
**Pull request by zadjii-msft**
I cannot for the life of me repro the original bug. I've got fonts with bad permissions SxS, I've tried installing a font twice, I've tried stopping the font cache service. No idea how to manually repro the original bug.

BUT theoretically, this function should never throw. So lets just switch this to a `LOG_IF_FAILED`, and hope that this goes away? 

* [x] Fixes #10211?
* [x] built & ran manually.

Unclear if this can get cherry-picked trivially to 1.8. Code's pretty trivial though so if we need another PR for that, it can be arranged.

### CODE REVIEW

**Review by miniksa — APPROVED** [review]

**Review by DHowett — CHANGES_REQUESTED** [review]
Can we also just make the font flalback loader non-fatal? There are a bunch of `THROW_IF_FAILED`s in `DxFontRenderData::NearbyCollection`, which contributes to this bug blowing up.

The caller, `DxFontReenderData::_FindFontFace` should treat failures coming out of  NearbyCollection as non-fatal -- you can probably wrap the entire "if the system collection missed" conditional in a try/`CATCH_LOG`.

**Comment by zadjii-msft** [comment]
@DHowett  I'm pretty those _are_ all in try/catch's. It's just they are all bubbling up to an HRESULT, getting here, and _this particular method_ is throwing it.

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
@miniksa may need to re-validate this w/ the new changes to Nearby

File: src/renderer/dx/DxFontRenderData.cpp [hunk 1]
```diff
@@ -644,63 +644,76 @@ CATCH_RETURN()
...
+    // move onto looking up a font from our hardcoded list of fonts
+    // that should really always be available.
+    if (!face)
+    {
+        for (const auto fallbackFace : FALLBACK_FONT_FACES)
+        {
+            familyName = fallbackFace;
+            // With these fonts, don't attempt the nearby lookup. We're looking
+            // for system fonts only. If one of the nearby fonts is causing us
+            // problems (like in GH##10211), then we don't want to go anywhere
```
Comment (miniksa) [inline src/renderer/dx/DxFontRenderData.cpp:697]:
```suggestion
            // problems (like in GH#10211), then we don't want to go anywhere

```

**Review by miniksa — APPROVED** [review]
Yeah I get it. That's about what I would have done. Thanks.

**Comment by miniksa** [comment]
@msftbot wait 6 minutes before merging

**Comment by greg-hammond** [comment]
follow link above to MS store download > Install > Launch > watch window crash within 10 seconds.
image

**Comment by zadjii-msft** [comment]
@greg-hammond Can you file a new issue, with a /feedback link in it? I'm guessing whatever crash you're seeing _isn't_ the same thing that was fixed by this PR 😉

**Pull request merged**